src/core/qgsvectorfilewriter.h

```diff
@@ -591,7 +593,8 @@ class CORE_EXPORT QgsVectorFileWriter : public QgsFeatureSink
                          QgsVectorFileWriter::FieldValueConverter *fieldValueConverter,
                          const QString &layerName,
                          QgsVectorFileWriter::ActionOnExistingFile action,
-                         QString *newLayer = nullptr
+                         QString *newLayer = nullptr,
+                         QgsCoordinateTransformContext transformContext = QgsCoordinateTransformContext()
```

Comment:
We really need to make these context arguments mandatory here, or we'll get unpredictable results depending on where files are created from (e.g. they are not actually set anywhere in this PR - so regardless of the user's settings you'll just get a default transform).